sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeFunctions.scala

```diff
@@ -54,3 +60,341 @@ case class CurrentTimestamp() extends LeafExpression {
     System.currentTimeMillis() * 1000L
   }
 }
+
+/**
+ * Abstract class for create time format expressions.
+ */
+abstract class TimeFormatExpression extends UnaryExpression with ExpectsInputTypes {
```

Comment:
I find this intermediate abstract class a bit excessive. I'm not sure how much benefits it provides. Can you just get rid of it?